pkg/cmd/template/render/parameters.go

```diff
@@ -0,0 +1,156 @@
+package render
+
+import (
+	"fmt"
+	"github.com/coreeng/corectl/pkg/cmdutil/userio"
+	"github.com/coreeng/corectl/pkg/template"
+	"gopkg.in/yaml.v3"
+	"os"
+	"slices"
+	"strings"
+)
+
+func CollectArgsFromAllSources(
+	spec *template.Spec,
+	argsFile string,
+	flagArgsRaw []string,
+	streams userio.IOStreams,
+	existingArgs []template.Argument,
+) ([]template.Argument, error) {
+	fileArgs, err := parseArgsFile(spec, argsFile)
+	if err != nil {
+		return nil, err
+	}
+
+	flagArgs, err := parseArgsFromFlags(spec, flagArgsRaw)
+	if err != nil {
+		return nil, err
+	}
+
+	passedArgs := make([]template.Argument, 0, len(fileArgs)+len(flagArgs))
+	passedArgs = append(passedArgs, fileArgs...)
+	passedArgs = append(passedArgs, flagArgs...)
+	for _, arg := range existingArgs {
```

Comment:
it's not deduplication logic, I'm filtering out arguments, which don't have corresponding parameters. It's different data structures, so I don't think it's applicable

There is kind of args deduplication logic right before the render, when all args are aggregated to map and passed to the templating library

Reply:
yeap cool, just was thinking to change data structure to Set, this way no duplicates are allowed, so if you insert same value twice it'll keep only single value

Reply:
In go there is no set built in (we would need to add another one dependency) + there is no real necessity for this, so I would leave it as is for now

Reply:
yeap, normally your do a map with empty/void struct that act as a set, so you'd need a new type. No worries, this might start a bigger change so this pr will grow.